src/shared/GitLab/GitLabAuthentication.cs

```diff
@@ -67,15 +70,69 @@ public async Task<AuthenticationPromptResult> GetAuthenticationAsync(Uri targetU
                 throw new ArgumentException(@$"Must specify at least one {nameof(AuthenticationModes)}", nameof(modes));
             }
 
-            if (Context.Settings.IsGuiPromptsEnabled && Context.SessionManager.IsDesktopSession &&
-                TryFindHelperCommand(out string helperCommand, out string args))
+            if (Context.Settings.IsGuiPromptsEnabled && Context.SessionManager.IsDesktopSession)
             {
-                return await GetAuthenticationViaHelperAsync(targetUri, userName, modes, helperCommand, args);
+                if (TryFindHelperCommand(out string helperCommand, out string args))
+                {
+                    return await GetAuthenticationViaHelperAsync(targetUri, userName, modes, helperCommand, args);
+                }
+
+                return await GetAuthenticationViaUiAsync(targetUri, userName, modes);
             }
 
             return GetAuthenticationViaTty(targetUri, userName, modes);
         }
 
+        private async Task<AuthenticationPromptResult> GetAuthenticationViaUiAsync(
+            Uri targetUri, string userName, AuthenticationModes modes)
+        {
+            var viewModel = new CredentialsViewModel(Context.Environment)
+            {
+                ShowBrowserLogin = (modes & AuthenticationModes.Browser) != 0,
+                ShowTokenLogin   = (modes & AuthenticationModes.Pat) != 0,
+                ShowBasicLogin   = (modes & AuthenticationModes.Basic) != 0,
+            };
+
+            if (!GitLabConstants.IsGitLabDotCom(targetUri))
+            {
+                viewModel.Url = targetUri.ToString();
+            }
+
+            if (!string.IsNullOrWhiteSpace(userName))
+            {
+                viewModel.UserName = userName;
+                viewModel.TokenUserName = userName;
+            }
+
+            await AvaloniaUi.ShowViewAsync<CredentialsView>(viewModel, GetParentWindowHandle(), CancellationToken.None);
+
+            if (!viewModel.WindowResult)
+            {
+                throw new Trace2Exception(Context.Trace2, "User cancelled dialog.");
```

Comment:
Is there a reason we can't use `ThrowIfWindowCancelled()` here?

Reply:
Good point! Updated.